dispatcher/src/main/java/io/zeebe/dispatcher/Dispatcher.java

```diff
@@ -191,10 +198,12 @@ private long offer(
 
         newPosition = updatePublisherPosition(activePartitionId, newOffset);
 
-        if (publisherPosition.proposeMaxOrdered(newPosition)) {
-          LOG.trace("Updated publisher position to {}", newPosition);
+        // if successful, replace position with simple counter
```

Comment:
If don't want to return -1 or -2 because we're at the end of the partition, then instead of returning a position based on the partitionId and the partitionOffset, we return a positions that's simply a count of records until now. How can I improve this? Or do you think it can be remove altogether?

Reply:
I was not sure whether this is an todo or it is related to the code. I think the comment is more confusing then helpful.

Reply:
```suggestion
        // if successful, replace internal publisher position with simple counter which will be returned
```